dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -45,6 +56,13 @@ internal object CoreFeature {
     internal var packageName: String = ""
     internal var packageVersion: String = ""
 
+    internal lateinit var dataUploadScheduledExecutor: ScheduledThreadPoolExecutor
+    internal lateinit var dataPersistenceExecutorService: ExecutorService
+
+    init {
+        initializeExecutors()
```

Comment:
why do you initialize it here and not in the `initialize` method below ?
Also, why do you re-initialize the executors in the `stop` method ?

Reply:
The thing is that if an executor is shut down it will never work again, you need reinitialise it but if I move the initialisation in the "initialize" method I don't need to reinitialise them in the "stop" method. I really don't know why I chose to initialise them in the "init" in the first place :)